src/commands/aksDeployManifest/aksDeployManifest.ts

```diff
@@ -0,0 +1,145 @@
+
+    if(!confirmed) {
+        vscode.window.showWarningMessage("Deployment cancelled.");
+        return;
+    }
+
+    const deploymentResult = await deployApplicationToCluster({
+        cluster: clusterPreference,
+        manifestPath: manifest
+    });
+
+    if(failed(deploymentResult)) {
+        vscode.window.showErrorMessage(deploymentResult.error);
+        return;
+    }
+
+    vscode.window.showInformationMessage(`Your applicatoin has been successfully deployed to the ${clusterPreference.clusterName} AKS cluster.`, "View resource in the Azure portal").then((res) => {
+        if(res) {
+            vscode.env.openExternal(vscode.Uri.parse(deploymentResult.result.url));
+        }
+    });
+}
+
+type ManifestQuickPickItem = vscode.QuickPickItem & {filePath: string};
+
+async function getManifestFile(): Promise<string | undefined> {
+    const items: ManifestQuickPickItem[] = [];
+    await vscode.workspace.findFiles(`**/**.yaml`, "**/node_modules/**").then((result) => {
+        result.forEach((fileUri) => {
+            const fileName = path.basename(fileUri.fsPath);
+            items.push({ label: fileName, description: fileUri.fsPath, filePath: fileUri.fsPath });
+        });
+    });
+
+    if(items.length === 0) {
+        vscode.window.showWarningMessage("No manifest files found in the workspace.");
+        return;
+    }
+
+    const fileSelected = await vscode.window.showQuickPick(items.sort(), {
+        title: "Select YAML",
+        placeHolder: "Select manifest to deploy",
+    });
+
+    if (!fileSelected) {
+        vscode.window.showWarningMessage("Manifest file not selected");
+        return;
+    }
+
+    return fileSelected.filePath;
+}
+
+async function confirmDeployment(clusterName: string): Promise<boolean> {
+    const confirmed = await vscode.window.showQuickPick(["Yes", "No"], {
+        title: `Do you want to deploy to this cluster: ${clusterName}?`,
+        placeHolder: "Select option",
+    });
+
+    return confirmed === "Yes";
+}
+
+type DeployApplicationToClusterOptions = {
+    cluster: ClusterPreference;
+    manifestPath: string;
+};
+type DeployApplicationToClusterResult = { url: string };
+
+async function deployApplicationToCluster(params: DeployApplicationToClusterOptions): Promise<Errorable<DeployApplicationToClusterResult>> {
+    const { cluster, manifestPath } = params;
+
+    const kubectl = await k8s.extension.kubectl.v1;
+
+    if (!kubectl.available) {
+        vscode.window.showWarningMessage(`Kubectl is unavailable.`);
+        return { succeeded: false, error: "Kubectl is unavailable." };
+    }
+
+    const kubeConfigFile = await createTempFile(cluster.kubeConfigYAML, "yaml");
+
+    const result = await longRunning(`Deployment of application to cluster: ${cluster.clusterName} in progress`, async () => {
+        return await invokeKubectlCommand(kubectl, kubeConfigFile.filePath, `apply -f ${manifestPath}`);
```

Comment:
💡 Note please: this long running already awaits for promise to be returned:

So, you can remove the return here

```.ts
    // Execute the deployment command
    const result = await longRunning(
        `Deploying application to cluster: ${cluster.clusterName} in progress...`,
        async () => invokeKubectlCommand(kubectl, kubeConfigFile.filePath, `apply -f ${manifestPath}`)
    );
```

Here is bit of a reconstruction of this function with comments, if this help thanks.

```*.ts
type DeployApplicationToClusterResult = { url: string };

async function deployApplicationToCluster(
    params: DeployApplicationToClusterOptions
): Promise<Errorable<DeployApplicationToClusterResult>> {
    const { cluster, manifestPath } = params;

    // Ensure kubectl is available
    const kubectl = await k8s.extension.kubectl.v1;
    if (!kubectl.available) {
        vscode.window.showWarningMessage("Kubectl is unavailable.");
        return { succeeded: false, error: "Kubectl is unavailable." };
    }

    // Create a temporary kubeconfig file
    const kubeConfigFile = await createTempFile(cluster.kubeConfigYAML, "yaml");

    // Execute the deployment command
    const result = await longRunning(
        `Deploying application to cluster: ${cluster.clusterName} in progress...`,
        async () => invokeKubectlCommand(kubectl, kubeConfigFile.filePath, `apply -f ${manifestPath}`)
    );

    // Check for errors during the kubectl command execution
    if (failed(result)) {
        return { succeeded: false, error: result.error };
    }

    // Generate and return the resource URL
    const resourceUrl = getPortalResourceUrl(getEnvironment(), cluster.clusterId);
    return { succeeded: true, result: { url: resourceUrl } };
}
```